web3/providers/legacy_websocket.py

```diff
@@ -43,6 +40,11 @@
     RPCResponse,
 )
 
+if typing.TYPE_CHECKING:
```

Comment:
If you import `TYPE_CHECKING` from the typing lib above this becomes: 
```suggestion
if TYPE_CHECKING:
```